src/module/item/base/document.ts

```diff
@@ -904,14 +905,15 @@ class ItemPF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extends Item
         data: this["_source"] | DeepPartial<this["_source"]>,
         action: DatabaseAction,
     ): Promise<boolean | void> {
+        const actor = this.actor;
+        if (!actor) return;
         // Look for any opportunity to abort early
         if (
             this.type !== "ancestry" &&
-            !data.system?.rules?.some((r) => ["BattleForm", "CreatureSize"].includes(r.key ?? ""))
+            !data.system?.rules?.some((r) => ["BattleForm", "CreatureSize"].includes(r.key ?? "")) &&
```

Comment:
This can probably be omitted if you're checking all the items anyways?

Reply:
As this can be called from `_preCreate` the triggering item could not exist in the item collection yet.